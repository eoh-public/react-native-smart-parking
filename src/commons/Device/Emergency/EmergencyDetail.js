import React, { memo, useCallback, useEffect } from 'react';
import { View, StyleSheet, Linking } from 'react-native';
import Text from '../../Text';
import { t } from 'i18n-js';
import MediaPlayer from '../../MediaPlayer';
import { standardizeCameraScreenSize } from '../../../utils/Utils';
import { Colors, Device } from '../../../configs';
import { RowUser } from '../../RowUser';
import { IconFill, IconOutline } from '@ant-design/icons-react-native';
import { useEmeragencyContacts } from '../../../screens/EmergencyContacts/hook';
import { useIsFocused } from '@react-navigation/core';
import _ from 'lodash';

const { standardizeHeight } = standardizeCameraScreenSize(
  Device.screenWidth + 86
);

const EmergencyDetail = memo(({ item }) => {
  const groupId = item.configuration.device.group.id;
  const isFocused = useIsFocused();
  const { listContacts, getListContacts } = useEmeragencyContacts();

  useEffect(() => {
    if (isFocused) {
      getListContacts(groupId);
    }
  }, [getListContacts, groupId, isFocused]);

  const onPressCall = useCallback((phoneNumber) => {
    Linking.openURL(`tel:${phoneNumber}`);
  }, []);
  return (
    <View style={styles.container}>
      <View style={styles.first}>
        <Text type="H4">{t('camera')}</Text>
        <MediaPlayer
          uri={_.get(item, 'configuration.uri', '')}
          previewUri={_.get(item, 'configuration.preview_uri', '')}
          style={styles.camera}
        />
      </View>
      <View style={styles.second}>
        <View>
          <Text type="H4">{t('emergency_contact')}</Text>
          {listContacts.map((contact, index) => (
            <RowUser
              key={contact.id.toString()}
              index={index}
              leftIcon={
                <IconOutline name={'user'} size={20} color={Colors.White} />
              }
              text={contact.name}
              rightComponent={
                <View style={styles.rightComponent}>
                  <IconFill
                    name={'phone'}
                    size={20}
                    color={Colors.Gray9}
                    onPress={() => onPressCall(contact.phone_number)}
                  />
                </View>
              }
            />
          ))}
        </View>
      </View>
    </View>
  );
});

export default EmergencyDetail;

const styles = StyleSheet.create({
  bottomButtonView: {
    position: 'absolute',
    bottom: 0,
  },
  container: {
    padding: 14,
    paddingTop: 0,
    marginBottom: 158,
    justifyContent: 'space-between',
  },
  camera: {
    height: standardizeHeight,
  },
  flatlistContent: {
    paddingHorizontal: 16,
  },
  center: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  linearView: {
    borderRadius: 95,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  text: {
    marginTop: 8,
    lineHeight: 32,
    fontSize: 24,
  },
  second: {
    marginTop: 24,
  },
  rightComponent: {
    flex: 1,
    justifyContent: 'center',
  },
});
