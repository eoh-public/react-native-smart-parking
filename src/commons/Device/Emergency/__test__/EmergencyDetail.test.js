import React from 'react';
import renderer, { act } from 'react-test-renderer';
import EmergencyDetail from '../EmergencyDetail';

jest.mock('@react-navigation/core', () => {
  return {
    ...jest.requireActual('@react-navigation/core'),
    useIsFocused: jest.fn(),
  };
});

describe('Test EmergencyDetail', () => {
  let tree;
  test('create EmergencyDetail', () => {
    const item = {
      configuration: {
        uri: '123',
        preview_uri: '123',
        device: {
          group: {
            id: 1,
          },
        },
      },
    };

    act(() => {
      tree = renderer.create(<EmergencyDetail item={item} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
