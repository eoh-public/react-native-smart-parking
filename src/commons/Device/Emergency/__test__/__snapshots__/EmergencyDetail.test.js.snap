// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EmergencyDetail create EmergencyDetail 1`] = `
<View
  style={
    Object {
      "justifyContent": "space-between",
      "marginBottom": 158,
      "padding": 14,
      "paddingTop": 0,
    }
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      Camera
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 10,
          "flex": 1,
          "justifyContent": "center",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Đang tải
        </Text>
      </View>
      <RCTVLCPlayer
        autoAspectRatio={true}
        autoplay={true}
        onVideoBuffering={[Function]}
        onVideoEnd={[Function]}
        onVideoEnded={[Function]}
        onVideoError={[Function]}
        onVideoLoadStart={[Function]}
        onVideoOpen={[Function]}
        onVideoPaused={[Function]}
        onVideoPlaying={[Function]}
        onVideoProgress={[Function]}
        onVideoStopped={[Function]}
        progressUpdateInterval={250}
        source={
          Object {
            "autoplay": true,
            "initOptions": Array [
              "--input-repeat=1000",
            ],
            "isNetwork": true,
            "uri": "123",
          }
        }
        src={
          Object {
            "isAsset": false,
            "isNetwork": true,
            "mainVer": 0,
            "patchVer": 0,
            "type": "",
            "uri": "123",
          }
        }
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "borderRadius": 10,
                "flex": 1,
              },
              Object {
                "height": 357,
              },
            ],
          ]
        }
        videoAspectRatio="21:9"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 24,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 16,
              "lineHeight": 24,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Danh sách liên hệ khẩn cấp
      </Text>
    </View>
  </View>
</View>
`;
